internal/host/instructionmanager.go

```diff
@@ -55,20 +56,23 @@ func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB, hwValidator hard
 	freeAddressesCmd := NewFreeAddressesCmd(log, instructionConfig.FreeAddressesImage)
 	resetCmd := NewResetInstallationCmd(log)
 	stopCmd := NewStopInstallationCmd(log)
+	dhcpAllocateCmd := NewDhcpAllocateCmd(log, instructionConfig.DhcpLeaseAllocatorImage, db)
 
 	return &InstructionManager{
 		log: log,
 		db:  db,
 		stateToSteps: stateToStepsMap{
-			HostStatusKnown:           {[]CommandGetter{connectivityCmd, freeAddressesCmd}, defaultNextInstructionInSec},
-			HostStatusInsufficient:    {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd}, defaultNextInstructionInSec},
-			HostStatusDisconnected:    {[]CommandGetter{inventoryCmd, connectivityCmd}, defaultBackedOffInstructionInSec},
-			HostStatusDiscovering:     {[]CommandGetter{inventoryCmd, connectivityCmd}, defaultNextInstructionInSec},
-			HostStatusPendingForInput: {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd}, defaultNextInstructionInSec},
-			HostStatusInstalling:      {[]CommandGetter{installCmd}, defaultBackedOffInstructionInSec},
-			HostStatusDisabled:        {[]CommandGetter{}, defaultBackedOffInstructionInSec},
-			HostStatusResetting:       {[]CommandGetter{resetCmd}, defaultBackedOffInstructionInSec},
-			HostStatusError:           {[]CommandGetter{stopCmd}, defaultBackedOffInstructionInSec},
+			models.HostStatusKnown:                    {[]CommandGetter{connectivityCmd, freeAddressesCmd, dhcpAllocateCmd}, defaultNextInstructionInSec},
+			models.HostStatusInsufficient:             {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd, dhcpAllocateCmd}, defaultNextInstructionInSec},
+			models.HostStatusDisconnected:             {[]CommandGetter{inventoryCmd, connectivityCmd}, defaultBackedOffInstructionInSec},
+			models.HostStatusDiscovering:              {[]CommandGetter{inventoryCmd, connectivityCmd}, defaultNextInstructionInSec},
+			models.HostStatusPendingForInput:          {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd, dhcpAllocateCmd}, defaultNextInstructionInSec},
+			models.HostStatusInstalling:               {[]CommandGetter{installCmd, dhcpAllocateCmd}, defaultBackedOffInstructionInSec},
+			models.HostStatusInstallingInProgress:     {[]CommandGetter{dhcpAllocateCmd}, defaultNextInstructionInSec},
```

Comment:
Note that during the installation the host get rebooted so this command might return an error.

Reply:
OK.  That is in the reply handling (inventory.go)

Reply:
It is called if get-next-step is called.  It is probably before the reboot.